packages/models-library/src/models_library/project_nodes.py

```diff
@@ -0,0 +1,188 @@
+from enum import Enum
+from typing import Dict, List, Optional, Union
+from uuid import UUID
+
+from pydantic import AnyUrl, BaseModel, Extra, Field, HttpUrl, constr, validator
+
+from .services import KEY_RE, PROPERTY_KEY_RE, VERSION_RE
+
+NodeID = UUID
+
+
+class RunningState(str, Enum):
+    UNKNOWN = "UNKNOWN"
+    PUBLISHED = "PUBLISHED"
+    NOT_STARTED = "NOT_STARTED"
+    PENDING = "PENDING"
+    STARTED = "STARTED"
+    RETRY = "RETRY"
+    SUCCESS = "SUCCESS"
+    FAILED = "FAILED"
+    ABORTED = "ABORTED"
+
+
+class Position(BaseModel):
+    x: int = Field(..., description="The x position", example=["12"])
+    y: int = Field(..., description="The y position", example=["15"])
+
+    class Config:
+        extra = Extra.forbid
+
+
+class AccessEnum(str, Enum):
+    ReadAndWrite = "ReadAndWrite"
+    Invisible = "Invisible"
+    ReadOnly = "ReadOnly"
+
+
+class PortLink(BaseModel):
+    nodeUuid: NodeID = Field(
```

Comment:
MINOR: I was wondering whether we should use ``node_uuid`` and set an alias. Let's keep this in mind for a model-library only PR

Reply:
yes we totally should